### ISSUE TITLE
serial connection fails with "Module did not self-register... ERR_DLOPEN_FAILED"

### ISSUE THREAD

**Issue by spidgrou** [SignalK/signalk-server#1355]
I'm trying to install a backup gps antenna on my OpenPlotter that works super well.
Unfortunately, I'm not able to do this simple installation.

I add this new connection selecting the port (specific port or openplotter, try both).
restart signal k
In the dashboard I can see an error.
In the server error I found:

```
Nov 09 16:00:18 Error: Module did not self-register: '/usr/lib/node_modules/signalk-server/node_modules/@serialport/bindings/build/Release/bindings.node'. at Object.Module._extensions..node (internal/modules/cjs/loader.js:1144:18) at Module.load (internal/modules/cjs/loader.js:950:32) at Function.Module._load (internal/modules/cjs/loader.js:790:12) at Module.require (internal/modules/cjs/loader.js:974:19) at require (internal/modules/cjs/helpers.js:93:18) at bindings (/usr/lib/node_modules/signalk-server/node_modules/bindings/bindings.js:112:48) at Object.<anonymous> (/usr/lib/node_modules/signalk-server/node_modules/@serialport/bindings/lib/linux.js:2:36) at Module._compile (internal/modules/cjs/loader.js:1085:14) at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10) at Module.load (internal/modules/cjs/loader.js:950:32) at Function.Module._load (internal/modules/cjs/loader.js:790:12) at Module.require (internal/modules/cjs/loader.js:974:19) at require (internal/modules/cjs/helpers.js:93:18) at Object.<anonymous> (/usr/lib/node_modules/signalk-server/node_modules/@serialport/bindings/lib/index.js:14:22) at Module._compile (internal/modules/cjs/loader.js:1085:14) at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10) at Module.load (internal/modules/cjs/loader.js:950:32) at Function.Module._load (internal/modules/cjs/loader.js:790:12) at Module.require (internal/modules/cjs/loader.js:974:19) at require (internal/modules/cjs/helpers.js:93:18) at Object.<anonymous> (/usr/lib/node_modules/signalk-server/node_modules/serialport/lib/index.js:2:17) at Module._compile (internal/modules/cjs/loader.js:1085:14) at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10) at Module.load (internal/modules/cjs/loader.js:950:32) at Function.Module._load (internal/modules/cjs/loader.js:790:12) at Module.require (internal/modules/cjs/loader.js:974:19) at require (internal/modules/cjs/helpers.js:93:18) at Object.<anonymous> (/usr/lib/node_modules/signalk-server/node_modules/@signalk/streams/serialport.js:63:20) { code: 'ERR_DLOPEN_FAILED' }
Nov 09 16:00:18 Error: Module did not self-register: '/usr/lib/node_modules/signalk-server/node_modules/@serialport/bindings/build/Release/bindings.node'. at Object.Module._extensions..node (internal/modules/cjs/loader.js:1144:18) at Module.load (internal/modules/cjs/loader.js:950:32) at Function.Module._load (internal/modules/cjs/loader.js:790:12) at Module.require (internal/modules/cjs/loader.js:974:19) at require (internal/modules/cjs/helpers.js:93:18) at bindings (/usr/lib/node_modules/signalk-server/node_modules/bindings/bindings.js:112:48) at Object.<anonymous> (/usr/lib/node_modules/signalk-server/node_modules/@serialport/bindings/lib/linux.js:2:36) at Module._compile (internal/modules/cjs/loader.js:1085:14) at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10) at Module.load (internal/modules/cjs/loader.js:950:32) at Function.Module._load (internal/modules/cjs/loader.js:790:12) at Module.require (internal/modules/cjs/loader.js:974:19) at require (internal/modules/cjs/helpers.js:93:18) at Object.<anonymous> (/usr/lib/node_modules/signalk-server/node_modules/@serialport/bindings/lib/index.js:14:22) at Module._compile (internal/modules/cjs/loader.js:1085:14) at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10) at Module.load (internal/modules/cjs/loader.js:950:32) at Function.Module._load (internal/modules/cjs/loader.js:790:12) at Module.require (internal/modules/cjs/loader.js:974:19) at require (internal/modules/cjs/helpers.js:93:18) at Object.<anonymous> (/usr/lib/node_modules/signalk-server/node_modules/serialport/lib/index.js:2:17) at Module._compile (internal/modules/cjs/loader.js:1085:14) at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10) at Module.load (internal/modules/cjs/loader.js:950:32) at Function.Module._load (internal/modules/cjs/loader.js:790:12) at Module.require (internal/modules/cjs/loader.js:974:19) at require (internal/modules/cjs/helpers.js:93:18) at Object.<anonymous> (/usr/lib/node_modules/signalk-server/node_modules/@signalk/streams/serialport.js:63:20) { code: 'ERR_DLOPEN_FAILED' }
Nov 09 16:00:18 signalk-server running at 0.0.0.0:[object Object]
Nov 09 16:00:18 /home/pi/.signalk/resources - OK...
```

Any help is welcome.

Thanks

**Comment by tkurki** [comment]
I think there is something wrong with your serialport module installation. I don't use OpenPlotter myself, but I think you should be able to reinstall SK server by running

`sudo npm install -g --unsafe-perm signalk-server`

**Comment by tkurki** [comment]
Have you updated your NodeJs version? What version are you running? `node --version`

**Comment by spidgrou** [comment]
@tkurki yes, I forgot to report my setup:
- NodeJs is 14.18.1
- NPM is 8.1.3

I will try to reinstall SK Server as suggested, but I need to do a backup first.
I'll tell you soon if something change.
Thanks for your fast answer, I really appreciated it.

**Comment by tkurki** [comment]
Internet hints that this is probably related to node js version not being in sync with the binary bindings that serialport needs to use to work with a specific platform's serial ports.

**Comment by spidgrou** [comment]
@tkurki just tried to execute the command but it doesn't work, it ask a previous version of node.js, the 10.
So I installed the node.js 10,24.1.
Then I installed again signalk-server with the command you suggested.
After a while a reboot but I have more other error, influxdb stop working...
So i restore my backup and remove my backup antenna.
As soon as I have time I reinstall a new Pi and try again.

Thanks for your time.
